lib/cacheman.ex

```diff
@@ -226,14 +259,38 @@ defmodule Cacheman do
     end
   end
 
+  def handle_call({:get_batch, keys}, _from, opts) do
+    response =
+      apply(opts.backend_module, :get_batch, [
+        opts.backend_pid,
+        Enum.map(keys, fn key ->
+          fully_qualified_key_name(opts, key)
+        end)
+      ])
+
+    case response do
+      {:ok, vals} ->
+        {:reply, {:ok, vals}, opts}
+
+      e ->
+        Logger.error("Cacheman - #{inspect(e)}")
+        {:reply, {:ok, :lists.concat(List.duplicate([nil], length(keys)))}, opts}
+    end
+  end
+
   def handle_call({:exists?, key}, _from, opts) do
     response =
       apply(opts.backend_module, :exists?, [
         opts.backend_pid,
         fully_qualified_key_name(opts, key)
       ])
 
-    {:reply, response, opts}
+    if is_boolean(response) do
+      {:reply, response, opts}
+    else
+      Logger.error("Cacheman - #{inspect(response)}")
+      {:reply, false, opts}
```

Comment:
🤔 Shouldn't we `raise` here ? We can't really tell if the key exists or not if the backend failed to give us this answer.

Reply:
Yeah, I would agree with you, but `get` uses the same logic: If redis isn't reachable, return nil, as if the key does not exist. And, based on tests that already exist, it seems that it is expected from exists? as well to return false, as if the key does not exist. Another problem with `raise` is the fact that it would terminate entire GenServer instance if not handled. That could potentially be a problem, if redis isn't reachable only momentary.

Does it make sense to return `{:error, description}`, like it is done in the `put` function, and to change `get` to return same :error tuple as well instead of `{:ok, nil}` ?

Reply:
🙇🏻  thanks for the explanation. This change can break some existing code that already uses ex-cacheman. Let's stick with the current behavior for now.